src/client/src/orangehrmRecruitmentPlugin/components/ViewActionHistory.vue

```diff
@@ -0,0 +1,219 @@
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-form-row>
+          <oxd-grid :rows="2" :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item
+              class="orangehrm-save-candidate-page --span-column-2"
+            >
+              <oxd-input-field
+                v-model="action.notes"
+                :label="$t('general.notes')"
+                type="textarea"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-divider />
+        <oxd-form-actions>
+          <oxd-button
+            display-type="ghost"
+            :label="$t('general.back')"
+            @click="onBack"
+          />
+          <submit-button :label="$t('general.save')" />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {APIService} from '@/core/util/services/api.service';
+import {navigate} from '@/core/util/helper/navigation';
+import {formatDate, parseDate} from '@/core/util/helper/datefns';
+import useLocale from '@/core/util/composable/useLocale';
+import useDateFormat from '@/core/util/composable/useDateFormat';
+
+const actionModel = {
+  candidate: null,
+  vacancy: null,
+  hiringManager: null,
+  status: null,
+  performedAction: null,
+  performedDate: null,
+  performedBy: null,
+  notes: null,
+};
+
+export default {
+  name: 'ViewActionHistory',
+
+  props: {
+    candidateId: {
+      type: Number,
+      required: true,
+    },
+    historyId: {
+      type: Number,
+      required: true,
+    },
+  },
+
+  setup(props) {
+    const {locale} = useLocale();
+    const {jsDateFormat} = useDateFormat();
+
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      `/api/v2/recruitment/candidates/${props.candidateId}/history`,
+    );
+    return {
+      http,
+      locale,
+      jsDateFormat,
+    };
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      action: {...actionModel},
+    };
+  },
+  computed: {
+    getPerformedDate() {
+      return formatDate(
+        parseDate(this.action.performedDate),
+        this.jsDateFormat,
+        {locale: this.locale},
+      );
+    },
+  },
+  beforeMount() {
+    this.isLoading = true;
+    this.http
+      .get(this.historyId)
+      .then(response => {
+        const {data} = response.data;
+        this.action.candidate = `${data.candidate.firstName} ${data.candidate.middleName} ${data.candidate.lastName}`,
+        this.action.vacancy = data.vacancy.name,
+        this.action.hiringManager = `${data.vacancy.hiringManager.firstName}
+          ${data.vacancy.hiringManager.middleName} ${data.vacancy.hiringManager.lastName}`,
+        this.action.status = data.action.label;
+        this.action.performedDate = data.performedDate;
+        this.action.performedAction = data.action.label;
+        this.action.performedBy = `${data.performedBy.firstName}
```

Comment:
may have to handle past employee. check with @devishke-orange, he is adding a helper method

Reply:
Sure I will check with him